torch/utils/_cuda_trace.py

```diff
@@ -0,0 +1,75 @@
+import traceback
+import sys
+from typing import Callable, List, Tuple, Any
+import logging
+
+
+class CallbackRegistry:
+    def __init__(self, name: str):
+        self.name = name
+        self.callback_list: List[Callable[[Tuple[Any, ...]], None]] = []
+
+    def add_callback(self, cb: Callable[[Tuple[Any, ...]], None]) -> None:
+        self.callback_list.append(cb)
+
+    def fire_callbacks(self, *args: int) -> None:
+        logger = logging.getLogger(__name__)
```

Comment:
It's common to make the logger a global long-lived variable.

Reply:
Thank you for pointing this out, fixed.